src/test/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessorTests.java

```diff
@@ -367,27 +407,28 @@ public void testInferencesSimilarity_whenExceptionFromMLClient_ThenFail() {
         Mockito.verifyNoMoreInteractions(singleSentenceResultListener);
     }
 
-    public void testInferenceSimilarity_whenNodeNotConnectedException_ThenTryThreeTimes() {
-        final NodeNotConnectedException nodeNodeConnectedException = new NodeNotConnectedException(
-            mock(DiscoveryNode.class),
-            "Node not connected"
-        );
+    public void testInferenceSentences_whenSpecificException_thenRetry() {
```

Comment:
just curious: what is the difference between this test and `testInferenceSentences_whenNodeNotConnectedException_thenRetry`? Seems like this test is also using `NodeNotConnectedException` for testing

Reply:
Yes, they are duplicated, we don't need add the new one agin for this case. Updated.